### ISSUE TITLE
Popups do not open when clicked

### ISSUE THREAD

**Issue by bsudekum** [nitaliano/react-native-mapbox-gl#102]
Even when hard coding a point seen below, the popup still does not open when clicked

``` java
view.addMarker(new MarkerOptions()
  .position(new LatLng(48.13863, 11.57603))
  .title("Hello World!")
  .snippet("Welcome to my marker."));
```

/cc @bleege

**Comment by Iragne** [comment]
any news on it ?

**Comment by bsudekum** [comment]
Stranger yet, the annotations are only clickable after rotating the screen.

**Comment by bleege** [comment]
Is this using version `2.3.0`?

**Comment by bsudekum** [comment]
Comment https://github.com/mapbox/react-native-mapbox-gl/issues/102#issuecomment-165865530 is using 2.3.0

**Comment by bleege** [comment]
@bsudekum Thanks for confirming.  Are other Gestures (double tap, fling, long press, etc) functioning as expected or are they not responding either?  The reason I ask is that the `MapView` [selects the Marker and opens the InfoWindow](https://github.com/mapbox/mapbox-gl-native/blob/55ff795672224abe94130b23f055f43801af3f75/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/MapView.java#L2650) inside it's [primary single finger Gesture listener object](https://github.com/mapbox/mapbox-gl-native/blob/55ff795672224abe94130b23f055f43801af3f75/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/MapView.java#L2524-L2729).

**Comment by erwan** [comment]
Same problem here with 3.0.0, with an "annotation" property I have the marker but no InfoWindow on click. Zooming, rotating, etc work. Even after rotation I still have to info window.

**Comment by erwan** [comment]
Do anyone has it working, and have clickable annotations on Android?

**Comment by ingwet** [comment]
Facing same problems, would appreciate any feedback on that

**Comment by tobrun** [comment]
Looking into this:

Currently noticing this during debugging:
- `OnMarkerClicked` callback is not invoked when we click a marker. 
- `OnMapLongClick` callback is invoked when we long click the map.

I will have to debug deeper in the Android SDK to see what is going wrong.

**Comment by tobrun** [comment]
During a second round of debugging, I'm noticing that `OnMarkerClicked` callback is invoked just once and afterwards never again since that marker remains in the list of `mSelectedMarkers`. I was able to debug to the code for showing an InfoWindow. I need to investigate why this is not showing, I initially thought the problem would be related to click handling.

**Comment by tobrun** [comment]
To make it even more weird:

This happens when you click an annotation and rotate the device:
![screen shot 2016-01-21 at 12 22 32](https://cloud.githubusercontent.com/assets/2151639/12479231/0cc0b1cc-c03a-11e5-9753-5e6fe79d84ec.png)

Even more strange if deselected and reactivated:
![screen shot 2016-01-21 at 12 22 40](https://cloud.githubusercontent.com/assets/2151639/12479242/27137a0a-c03a-11e5-9ada-4f6f23377722.png)

**Comment by tobrun** [comment]
The reason why it's is shown the second time will probably be related to this method in onCreate:

``` java
 private void adjustTopOffsetPixels() {
        int count = mAnnotations.size();
        for (int i = 0; i < count; i++) {
            Annotation annotation = mAnnotations.get(i);
            if (annotation instanceof Marker) {
                Marker marker = (Marker) annotation;
                marker.setTopOffsetPixels(
                        getTopOffsetPixelsForSprite(marker.getIcon()));
            }
        }

        for (Marker marker : mSelectedMarkers) {
            if (marker.isInfoWindowShown()) {
                Marker temp = marker;
                temp.hideInfoWindow();
                temp.showInfoWindow();
                marker = temp;
            }
        }
    }
```

Edit: 
I have tried executing this method with a delayed handler using reflection but this didn't show the InfoWindow as I suspected it would. 

Edit 2:
I'm now noticing the react native configuration for handling orientation which is by default onConfigChanges which will not trigger a call to onCreate.

**Comment by tobrun** [comment]
I'm going to try 2 things now, 
- Properly configure MapView with the different lifecycle events 
  - Edit: this is not possible, ReactNative does its config after onResume
- Debug ReactNative core code to see what happens with orientation change 
  - Edit: orientation change is not recreating the Activity -> configChanges config

**Comment by bsudekum** [comment]
@tobrun ref activity docs: http://facebook.github.io/react-native/docs/native-modules-android.html#getting-activity-result-from-startactivityforresult

**Comment by gundersen** [comment]
@tobrun @zugaldia -- how important is  opening annotation popups before we blog this? cc @bsudekum

**Comment by tobrun** [comment]
@gundersen
I feel it's pretty important, I'm not confident calling this a v1.0 release because I'm unable to give you an estimation for the solution or the impact it will have on current architecture. If you want to blog I'm suggesting calling the Android SDK a _beta_ or _pre-release_ version. 

@bsudekum 
Next steps to resolve this issue would be to reimplementing the Android code based on this [GoogleMaps ReactNative library](https://github.com/lelandrichardson/react-native-maps). Currently my schedule is being prioritised in favour of Android 4.0.0 release work. 

cc @zugaldia

**Comment by zugaldia** [comment]
@gundersen @tobrun per voice with @bsudekum today there're two significant blocker issues before we're ready for prime time. This and https://github.com/mapbox/mapbox-gl-native/issues/3584. We're looking into them.

**Comment by zugaldia** [comment]
Related: https://github.com/mapbox/mapbox-gl-native/issues/3584#issuecomment-179428362

**Comment by jayfunk** [comment]
I am still experiencing this issue in the latest release. https://github.com/mapbox/react-native-mapbox-gl/issues/293 It appears that if I add another element which changes the map dimensions the popup displays but it will render the popup in the incorrect position depending on how much the map is resized.

**Comment by ghost** [comment]
@bsudekum  Will #313 be merged any time soon? I was able to fix this issue on mapboxgl 4.0

I am currently working on allowing react components to be rendered as InfoWindows but once done I will make a new pull request for this issue.

![screencast](https://cloud.githubusercontent.com/assets/16978609/15346378/5547b1d2-1c6c-11e6-9afe-6b01a6534956.gif)

**Comment by bsudekum** [comment]
@AlexLourenco this is awesome! I have a local branch that incorporates v4 but I was not able to fix this bug, so no plans to merge any time soon.

**Comment by dominicmh** [comment]
For now, the only way to get this working is by registering to the `DeviceEventEmitter` like so:
`DeviceEventEmitter.addListener('onOpenAnnotation', this.onOpenAnnotation);`

Is that right or did I miss something?

**Comment by afischer** [comment]
Any word on this? Currently having this issue with the latest releases.

**Comment by dapetcu21** [comment]
I implemented a workaround for this issue in 5.0.0. It's now fixed in master. Every time an annotation gets clicked, I manually trigger a relayout of the annotation view.
